src/Services/MiningRewardsService.ts

```diff
@@ -27,16 +26,21 @@ export class MiningRewardsService {
     }
 
     public async fetchMiningRewards(nodeId: number, filter: string) {
-        return await MiningReward.findAll({
-            raw: true,
-            where: {
-                nodeId,
-                updatedAt: {
-                    [Op.gte]:
-                        moment.utc().subtract(1, filter as unitOfTime.Base).format("YYYY-MM-DD HH:MM:ssZZ")
-                }
-            },
-            order: [['updatedAt', 'DESC']]
-        });
+        return await datebase.runQuery<MiningReward>(
+            `select date_trunc(:period, "updatedAt") "timePeriod", sum("rewardAmount") "rewardSum"
+            from "MiningRewards"
+            where "nodeId" = :nodeId
+            and "updatedAt" >= now() - interval :filter
+            group by date_trunc(:period, "updatedAt")
+            order by "timePeriod" desc;`,
+            {
+                replacements: {
+                    filter: `1 ${filter}`,
+                    nodeId: nodeId,
+                    period: filter == "day" ? "hour" : "day"
```

Comment:
Can you extract this into a function somewhere? In case we need to change it it might be good to have it in one place so all filters behave the same for the period.

Reply:
I'll do, but not sure about it because Sequelize doesn't like and is very strict about things like that.